app/controllers/contests_controller.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class ContestsController < ApplicationController
+  before_action :authenticate_user!, except: [:index]
+  # before_action :authorize_admin, only: %i[admin]
+
+  # GET /contests
+  def index
+    @contests = Contest.all.order(id: :desc)
+  end
+
+  # GET /contests/:id
+  def show
+    @contest = Contest.find(params[:id])
+    @submissions = @contest.submissions
+    @user_count = User.count
+  end
+
+  # GET /contests/admin
+  def admin
+    @contests = Contest.all.order(id: :desc)
+  end
+
+  def close_contest
+    @contest = Contest.find(params[:id])
+    @contest.deadline = Time.zone.now
+    @contest.status = "Completed"
+    respond_to do |format|
+      if @contest.save
+        format.html { redirect_to contest_page_path(@contest.id), notice: "Contest was successfully ended." }
+        format.json { render :show, status: :created, location: @contest }
+      else
+        format.html { render :admin }
+        format.json { render json: @contest.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # POST /contest/create
+  def create
+    # checking for any other live contest, if found mark is as completed
+    Contest.all.each do |contest|
+      if contest.status == "Live"
```

Comment:
```ruby 
  if Contest.exists?(status: 'live')
    "Concurrent contests are not allowed, close other contests before creating a new one"
    end
 ```
 Add something like this instead of closing existing contests automatically